sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_container_client.hpp

```diff
@@ -257,6 +257,21 @@ namespace Azure { namespace Storage { namespace Blobs {
         const std::string& blobName,
         const DeleteBlobOptions& options = DeleteBlobOptions()) const;
 
+    /**
+     * @brief Creates a new block blob under this container. For partial block blob updates and
+     * other advanced features, please see BlockBlobClient. To create or modify page or see
+     * PageBlobClient or AppendBlobClient.
+     *
+     * @param blobName The name of the blob to create.
+     * @param content A BodyStream containing the content to upload.
+     * @param options Optional parameters to execute this function.
+     * @return A BlockBlobClient referencing the newly created block blob.
+     */
+    Azure::Core::Response<BlockBlobClient> UploadBlob(
```

Comment:
Why do we only have this for block blob? Is there anything special for block blob?

Reply:
I will redirect this question to @tg-msft to answer.

Reply:
We made the choice in other languages to prioritize block blobs because that's what most customers want.  Especially when they're getting started.  Step 0 of working with many Azure services is uploading your data to Storage and then giving that service a SAS URI.  Data scientists doing ML, physicists playing with Quantum, etc., don't really want to dig deeply into all of Storage's features and different types of blobs.  They just want to get their data into Storage as quickly as possible and move on to the next step.  Storage block blobs power the rest of Azure so we're guiding customers to them.